source/readonly-deep.d.ts

```diff
@@ -0,0 +1,57 @@
+import {Primitive} from './basic';
+
+/**
+Convert `object`s, `Map`s, `Set`s, and `Array`s into immutable stuctures recursively. Converts all properties/contents into immutable structures recusively.
+
+This is useful when a deeply nested structure needs to be exposed as completely immutable, for example, an imported JSON module.
+
```

Comment:
Maybe add a note to encourage people to upvote https://github.com/microsoft/TypeScript/issues/13923 if they want to see this type as a built-in?